### PR TITLE
[stable/wordpress] Allow more flexible TLS configuration.

### PR DESCRIPTION
**Pull request by adtennant**
#### What this PR does / why we need it:
Allows more flexible TLS configuration. Currently you can't do certain things, e.g. generate certificates that cover multiple hosts. Instead of having a custom structure for TLS this PR reverts to using the standard Ingress structure.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

File: stable/wordpress/values.yaml [hunk 1]
```diff
@@ -224,11 +224,12 @@ ingress:
...
     path: /
 
-    # Set this to true in order to enable TLS on the ingress record
-    tls: false
-
-    ## If TLS is set to true, you must declare what secret will store the key/certificate for TLS
-    tlsSecret: wordpress.local-tls
+  ## The tls configuration for the ingress
+  ## see: https://kubernetes.io/docs/concepts/services-networking/ingress/#tls
+  tls:
```
Comment (juan131) [inline stable/wordpress/values.yaml:234]: I'd add another boolean parameter "tls.enabled" that allows you specify whether to enable/disable TLS in the Ingress definition.

File: stable/wordpress/README.md [hunk 2]
```diff
@@ -99,8 +99,9 @@ The following table lists the configurable parameters of the WordPress chart and
 | `ingress.annotations`            | Ingress annotations                        | `[]`                                                    |
 | `ingress.hosts[0].name`          | Hostname to your Wordpress installation    | `wordpress.local`                                       |
 | `ingress.hosts[0].path`          | Path within the url structure              | `/`                                                     |
+| `ingress.tls[0].hosts[0]`        | TLS hosts                                  | `wordpress.local`                                       |
+| `ingress.tls[0].secretName`      | TLS Secret (certificates)                  | `wordpress.local-tls`                                   |
 | `ingress.hosts[0].tls`           | Utilize TLS backend in ingress             | `false`                                                 |
```
Comment (juan131) [inline stable/wordpress/README.md]: Shouldn't you remove this parameter?

**Review by juan131 — CHANGES_REQUESTED** [review]
Hi @adtennant 

Please take a look to my comments! Please also notice that you need to sign every commit or the DCO tests will fail.

File: stable/wordpress/values.yaml (hunk 1 above)
Comment (adtennant) [reply stable/wordpress/values.yaml:234]: I feel that's redundant as a blank or missing `tls` section in the ingress indicates that intent and matches up with what other charts do, e.g. `stable/artifactory` (there are many more a quick search for `{{ toYaml .Values.ingress.tls | indent 4 }}` in the repo will find them)

Comment (juan131) [reply stable/wordpress/values.yaml:234]: Okay, let's keep it your way. I'm okay with this approach.

**Review by juan131 — APPROVED** [review]

**Comment by juan131** [comment]
Could you please rebase from master and bump the version of the chart?

**Comment by juan131** [comment]
Hi @adtennant 

I just merged your other PR to unset the nodePort and, as a consequence, the CI complains again about conflicts in the Chart.yml. could you rebase agin?

**Pull request merged**